ui.frontend/src/view/FormPanel.js

```diff
@@ -180,6 +180,20 @@ class FormPanel extends FormFieldBase {
         this.element.setAttribute(Constants.DATA_ATTRIBUTE_VALID, valid);
     }
 
+    /**
+     * Updates the HTML state based on the label state of the field.
+     * @param {Object} label - The label.
+     */
+    updateLabel(label) {
+        super.updateLabel(label);
+
+        const bemClass = Array.from(this.element.classList).filter(bemClass => !bemClass.includes('--'))[0];
+        const labelContainer = this.element.querySelector(`.${bemClass}__label-container`);
+        if (labelContainer) {
+            labelContainer.querySelector("label").setAttribute(Constants.ROLE, "heading");
```

Comment:
role="heading" is not required on a label container. Labels are already accessible using the for attribute and the corresponding id.